src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -266,7 +266,7 @@ public boolean checkSolidity(Hash hash, boolean milestone, int maxProcessedTrans
 
                         if (!transactionRequester.isTransactionRequested(hashPointer, milestone)) {
                             transactionRequester.requestTransaction(hashPointer, milestone);
-                            break;
+                            continue;
```

Comment:
not sure if the following suggestion makes sense:
if it's a milestone, then `continue`, else `break`.

the reason being:
this mechanism can be used to fill the request queue with arbitrary data faster.
```
0
| \
0  ?
0
| \
0  ?
0
| \
0  ?
...
```
in the current code the first `?` would be requested and stop (and if this `?` does not exist the rest will never be requested, with this change all the historical `?`s will be requested (up until the `maxProccessedTransactions` is reached).

I'm not sure this is such an issue, but my proposal would not change behaviour for random transactions, but acheive the improvement for syncing nodes.